File: frontend/src/core/types/crime3d.ts
export enum EvidenceMarkerType {
    WEAPON = "weapon",
    BLOOD = "blood",
    FINGERPRINT = "fingerprint",
    FOOTPRINT = "footprint",
    OTHER = "other"
}

export interface EvidenceMarker {
    id: string;
    type: EvidenceMarkerType;
    description: string;
    position_x: number;
    position_y: number;
    position_z: number;
    image_url?: string;
    collected_by?: string;
    timestamp: string;
}

export interface CrimeScene {
    id: string;
    case_id: string;
    location: string;
    date: string;
    description: string;
    markers: EvidenceMarker[];
    model_url?: string;
    metadata: Record<string, any>;
}

export interface CreateSceneRequest {
    case_id: string;
    location: string;
    date: string;
    description: string;
}

export interface AddMarkerRequest {
    type: EvidenceMarkerType;
    description: string;
    position_x: number;
    position_y: number;
    position_z: number;
}

File: backend/app/schemas/crime3d.py
"""
Crime Scene 3D Schemas - Skill 17
3D Reconstruction and Evidence Mapping
"""
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import date, datetime
from enum import Enum


class EvidenceMarkerType(str, Enum):
    WEAPON = "weapon"
    BLOOD = "blood"
    FINGERPRINT = "fingerprint"
    FOOTPRINT = "footprint"
    OTHER = "other"


class EvidenceMarker(BaseModel):
    id: str
    type: EvidenceMarkerType
    description: str
    position_x: float
    position_y: float
    position_z: float
    image_url: Optional[str] = None
    collected_by: Optional[str] = None
    timestamp: datetime


class CrimeScene(BaseModel):
    id: str
    case_id: str
    location: str
    date: date
    description: str
    markers: List[EvidenceMarker] = []
    model_url: Optional[str] = None # URL to 3D model file (GLB/GLTF)
    metadata: Dict[str, Any] = {}


class CreateSceneRequest(BaseModel):
    case_id: str
    location: str
    date: date
    description: str


class AddMarkerRequest(BaseModel):
    type: EvidenceMarkerType
    description: str
    position_x: float
    position_y: float
    position_z: float
